src/AwkwardArray.jl

```diff
@@ -1,6 +1,6 @@
 module AwkwardArray
 
-abstract type Content end
+abstract type Content <: AbstractArray{undef,1} end
```

Comment:
```suggestion
abstract type Content{T} <: AbstractArray{T,1} end
```

could work, if the `Content` has a well-defined concepts of element type `eltype{T}`, which I think it does not.

I DID NOT know you can use `undef` like this!

Reply:
In the long run, you can't. I didn't know what I was doing.

I think I have better types in e74b5b4e5d9232482d8abc802eb6911e17d8960b. All of the struct fields are now parameterized, rather than generic. (If instantiated with concrete parameter values, everything should be concrete and fast.)

Reply:
Sounds good we can resolve this conversation.

Reply:
This eventually became

```julia
abstract type Content <: AbstractArray{T where T<:Any,1} end
```

because I didn't want to say what the item type of `Content` is at the level of these definitions, though when it's instantiated, it should be concrete. (I hope.)

Reply:
if you have function barrier between `func(c::Content)` and the actual heavy computation using individual "columns" of `c`, you might still be fine, and `Content` can stay an abstract type maybe